video/out/vo_gpu_next.c

```diff
@@ -1547,10 +1547,15 @@ static void save_cache_files(struct priv *p)
 
         if (!target_file)
             continue;
-        char *tmp = talloc_asprintf(ta_ctx, "%s~", target_file);
-        FILE *cache = fopen(tmp, "wb");
-        if (!cache)
+        char *tmp = talloc_asprintf(ta_ctx, "%sXXXXXX", target_file);
+        int fd = mkstemp(tmp);
```

Comment:
not sure why we have this wrapper but consider using `mp_mkostemps`

Reply:
No, sorry, I'm not going to use some NIHed version with `fuckmess` and `crap` in it without good reason. 

> Oh, and my wrapper is probably buggy. Fortunately that doesn't matter.

The whole wm4 meltdown over O_CLOEXEC doesn't matter here. And I'm not going into the reset...

Reply:
why doesn't O_CLOEXEC matter here actually? some other thread could spawn a process in the same moment and then leak the fd

Reply:
> why doesn't O_CLOEXEC matter here actually? some other thread could spawn a process in the same moment and then leak the fd

It matters in demux case, because by default file is unlinked and we want for it to be removed as soon as mpv is closed.

In wm4 own words:
> The fact that the file, which is expected to reach double or tripple digit GB sizes, will be deleted only once all processes unreference the FD, makes this sort of a big deal.

In our case it is not that critical. Also the time window is very small compared to demux cache which lives for whole duration of playback (if enabled).

I agree leaking fd may be not great, but it is not a problem in this cache case. And we don't even want this file to be removed, we rename it.